### PR TITLE
For #36439, 0.18 compatible session cache

### PR DESCRIPTION
**Pull request by jfboismenu**
Makes the session cache from the desktop compatible with the 0.18 one.

### CODE REVIEW

File: python/shotgun_desktop/startup.py [hunk 1]
```diff
@@ -191,13 +192,22 @@ def __initialize_sgtk(sgtk, app_bootstrap):
...
 
-        # get the current user
-        user = sg_auth.get_default_user()
-        logger.info("Setting current user: %r" % user)
-        sgtk.set_authenticated_user(user)
+        # get the current user for the pipeline.
+        core_user = sg_auth.get_default_user()
+
+        # If there is a user and it's a script user, use that
+        if core_user and not core_user.login:
```
Comment (manneohrstrom) [inline python/shotgun_desktop/startup.py]: no possibility that `core_user` is a script user? I am guessing not - just double checking...

File: python/shotgun_desktop/startup.py [hunk 2]
```diff
@@ -191,13 +192,22 @@ def __initialize_sgtk(sgtk, app_bootstrap):
         dm = sgtk.util.CoreDefaultsManager()
         sg_auth = shotgun_authentication.ShotgunAuthenticator(dm)
 
-        # get the current user
-        user = sg_auth.get_default_user()
-        logger.info("Setting current user: %r" % user)
-        sgtk.set_authenticated_user(user)
+        # get the current user for the pipeline.
```
Comment (manneohrstrom) [inline python/shotgun_desktop/startup.py]: i feel there is no explanation here about core versions and why we need to do this - i think that could be helpful for future code readers.

File: python/tk-core/python/tank_vendor/shotgun_authentication/session_cache.py [hunk 3]
```diff
@@ -83,6 +83,20 @@ def _get_site_cache_location(base_url):
...
+def _get_018_site_cache_location(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+
+    :param base_url: Site we need to compute the root path for.
+
+    :returns: An absolute path to the site cache root.
+    """
+    return os.path.join(
+        _get_cache_location(),
```
Comment (manneohrstrom) [inline python/tk-core/python/tank_vendor/shotgun_authentication/session_cache.py]: 😢  no way we can use core to figure out this location?

**Comment by manneohrstrom** [comment]
cr looks good - my main worry is that we are (again) duplicating core logic in desktop. Any way we can avoid it? Is this interim until we upgrade desktop to 0.18 - then i think we should add TODOs to point this out.

File: python/shotgun_desktop/startup.py (hunk 1 above)
Comment (jfboismenu) [inline python/shotgun_desktop/startup.py]: the `if` does test for the user being a script user.

Comment (manneohrstrom) [inline python/shotgun_desktop/startup.py]: sorry i was being daft! :facepalm:

**Pull request closed without merging**